proto/src/determined/checkpoint/v1/checkpoint.proto

```diff
@@ -30,45 +22,68 @@ enum State {
   STATE_DELETED = 4;
 }
 
-// Checkpoint is an artifact created by a trial during training.
+// Aggregate and batch metrics
+message Metrics {
+  option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
+    json_schema: { required: [ "avg_metrics" ] }
+  };
+  // Aggregate user-generated metrics
+  google.protobuf.Struct avg_metrics = 1;
+  // User-generated metrics for each batch
+  repeated google.protobuf.Struct batch_metrics = 2;
+}
+
+// CheckpointTrainingMetadata is specifically metadata about training.
+message CheckpointTrainingMetadata {
+  option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
+    json_schema: { required: [] }
```

Comment:
do you mean if required array is empty the whole `option` section should be skipped? yes we can remove it if you prefer.

Reply:
ah that doesn't matter much IMO. I was just checking if can make guarantees on any of those fields being present then we should otherwise the generated code treats all of them as nullable. eg experiment_id or trial_id